developer-docs-site/docs/tutorials/first-dapp.md

```diff
@@ -368,6 +368,217 @@ You can also verify that the module was published by going to the [Aptos Explore
 ```
 
 Make a note of `"name": "message"`, we will use it in the next section.
+</details>
+
+<details>
+<summary>Publish the `hello_blockchain` module with the TS SDK</summary>
+We will use the Aptos CLI to compile the `hello_blockchain` module and use the TS SDK to publish the module.
+
+1. Download [the `hello_blockchain` package](https://github.com/aptos-labs/aptos-core/tree/main/aptos-move/move-examples/hello_blockchain).
+
+2. Next, use the `aptos move compile --save-metadata` command (replacing `/path/to/hello_blockchain/` and `<address>`):
+
+```bash
+aptos move compile --save-metadata --package-dir /path/to/hello_blockchain/ --named-addresses hello_blockchain=<address>
+```
+
+For example:
+
+```bash
+aptos move compile --save-metadata --package-dir ~/code/aptos-core/aptos-move/move-examples/hello_blockchain/ --named-addresses hello_blockchain=0x5af503b5c379bd69f46184304975e1ef1fa57f422dd193cdad67dc139d532481
+```
+
+The `--named-addresses` replaces the named address `hello_blockchain` in `hello_blockchain.move` with the specified address. For example, if we specify `--named-addresses hello_blockchain=0x5af503b5c379bd69f46184304975e1ef1fa57f422dd193cdad67dc139d532481`, then the following:
+
+```rust
+module hello_blockchain::message {
+```
+
+becomes:
+
+```rust
+module 0x5af503b5c379bd69f46184304975e1ef1fa57f422dd193cdad67dc139d532481::message {
+```
+
+This makes it possible to publish the module for the given account (in this case our wallet account, `0x5af503b5c379bd69f46184304975e1ef1fa57f422dd193cdad67dc139d532481`).
+
+The `--save-metadata` saves the package metadata in the package's build directory. If set, package metadata should be generated and stored in the package's build directory. This metadata can be used to construct a transaction to publish a package.
+
+At this point, we should have a `build` folder in the same directory of our `hello_blockchain` folder. Next step would be to publish the module to the chain. 
+The TS SDK provides us a `publishPackage()` function where it expects to get both package metadata and the move module as `Uint8Array`. We can do it by converting both `package-metadata.bcs` file and the `bytecode_modules/message.mv` module into hex strings (using a command), and then to `Uint8Array` (using the SDK).
+
+Convert `package-metadata.bcs` file and the `bytecode_modules/message.mv` module into hex strings
+
+`cd` into the `hello_blockchain/build/Example` directory
+```bash
+cd hello_blockchain/build/Example
+```
+
+Convert `package-metadata.bcs` to a hex string. On Mac and Linux we can use the command
+```bash
+cat package-metadata.bcs | od -v -t x1 -A n | tr -d ' \n'
+```
+That will output a hex string we can later use.
+
+Convert `message.mv` to a hex string. On Mac and Linux we can use the command
+```bash
+cat bytecode_modules/message.mv | od -v -t x1 -A n | tr -d ' \n'
+```
+That will output a hex string we can later use.
+
```

Comment:
Should we just load these directly from the files into Typescript?  Might be easier than first converting to Hex, and then using them.

Reply:
yes we cam, same as I started to do in the ANS implementation https://github.com/aptos-labs/aptos-core/pull/6000/files#diff-af27ba8aea8ccbfc871dd0e005811cc7198ad93d52686f017e79bc15483c96bdR46-R58 
we still need to compile them first though - can use `aptos move compile` for that

Reply:
On a side note... I think there's a project that would make the SDKs work more seamlessly with the CLI
* Make local testing faster
* Add support for compilation and publishing
* Not need this interesting thing

Reply:
The TS SDK? Interesting..